### ISSUE TITLE
FileNotFoundException (Is a directory) with 2.4.11

### ISSUE THREAD

**Issue by mjmeintjes** [thheller/shadow-cljs#330]
Just upgraded to 2.4.11, and my build is now failing with the following error:
```
FileNotFoundException: /home/matthys/projects/personal/learning/node_modules/asn1.js (Is a directory)
        java.io.FileInputStream.open0 (FileInputStream.java:-2)
        java.io.FileInputStream.open (FileInputStream.java:196)
        java.io.FileInputStream.<init> (FileInputStream.java:139)
        clojure.java.io/fn--10968 (io.clj:229)
        clojure.java.io/fn--10968 (io.clj:229)
        clojure.java.io/fn--10881/G--10874--10888 (io.clj:69)
        clojure.java.io/fn--10942 (io.clj:165)
        clojure.java.io/fn--10942 (io.clj:165)
```

**Comment by jiyinyiyong** [comment]
Maybe we need more details like `shadow-cljs.edn`?

**Comment by mjmeintjes** [comment]
I'll try to isolate it as this is from a project with a big `shadow-cljs.edn` and `package.json`.

**Comment by mjmeintjes** [comment]
Some more info:
With 2.4.8, compilation works.

With 2.4.11, I get the above error.

With 2.4.10, I get:
```
The required JS dependency "process" is not available, it was required by "learning/web/nodejs_polyfill.cljs".

Searched in:/home/matthys/projects/personal/learning/node_modules

You probably need to run:
  npm install process

See: https://shadow-cljs.github.io/docs/UsersGuide.html#npm-install
```

With 2.4.9, I get the following error:
```
FileNotFoundException: /home/matthys/projects/personal/learning/node_modules/validate.io-undefined/lib (Is a directory)
        java.io.FileInputStream.open0 (FileInputStream.java:-2)
        java.io.FileInputStream.open (FileInputStream.java:196)
        java.io.FileInputStream.<init> (FileInputStream.java:139)
        clojure.java.io/fn--10968 (io.clj:229)
        clojure.java.io/fn--10968 (io.clj:229)
        clojure.java.io/fn--10881/G--10874--10888 (io.clj:69)
        clojure.java.io/fn--10942 (io.clj:165)
        clojure.java.io/fn--10942 (io.clj:165)
```

**Comment by mjmeintjes** [comment]
Here is my shadow-cljs.edn:

```
;; shadow-cljs configuration
{:source-paths
 ["src"
  "test"
  #_"keechma-toolbox/src/cljs"]

 :dependencies [[metosin/spec-tools "0.5.1"]
                [com.taoensso/timbre "4.10.0"]

                [orchestra "2017.11.12-1"]
                [expound "0.4.0"]
                [instaparse "1.4.8"]
                ;; [com.cerner/clara-rules "0.17.0"]
                ;; development/testing
                [devcards "0.2.4"]
                [data-frisk-reagent "0.4.5" :exclusions [cljsjs/react cljsjs/react-dom]]
                [philoskim/debux "0.4.1"]
                [com.taoensso/truss "1.5.0"]
                [hodgepodge "0.1.3"]
                ;; [debux.cs.core :as d :refer-macros [clog clogn dbg dbgn break]]

                [binaryage/devtools "0.9.7"]
                [org.clojure/spec.alpha "0.1.143"]
                [org.clojure/test.check "0.10.0-alpha2"]
                [org.clojars.mihaelkonjevic/cljs-react-test "0.1.5" :exclusions [cljsjs/react-with-addons]]

                ;; [cider/cider-nrepl "0.18.0-SNAPSHOT"]
                [cider/cider-nrepl "0.18.0-SNAPSHOT"]
                [refactor-nrepl "2.4.0-SNAPSHOT"]
                [vvvvalvalval/scope-capture "0.1.4"]
                [vvvvalvalval/scope-capture-nrepl "0.2.0"]

                ;; general
                [funcool/promesa "1.9.0"]
                [funcool/httpurr "1.0.0"]
                [com.taoensso/tufte "1.3.0"]
                [binaryage/oops "0.5.8"]
                ;;[funcool/cuerdas "2.0.4"] ;; funcool string manipulation
                [camel-snake-kebab "0.4.0"]
                [net.cgrand/xforms "0.16.0"] ;; extra transducer functionality
                [com.andrewmcveigh/cljs-time "0.5.2"]
                [clj-time "0.14.2"] ;;for macros

                [com.rpl/specter "1.1.1"]
                [funcool/lentes "1.2.0"]
                [org.martinklepsch/derivatives "0.3.0"]
                [prismatic/plumbing "0.5.5"]
                [com.cognitect/transit-cljs "0.8.243"]
                [rand-cljc "0.1.0"] ;; random with seed
                [com.taoensso/encore "2.89.0"]
                [bidi "2.1.2"]
                [com.cemerick/url "0.1.1"]


                ;; ui
                [reagent-material-ui "0.2.5"]
                [reagent "0.8.0"]

                [keechma "0.3.6" :exclusions [keechma/entitydb cljsjs/react cljsjs/react-dom]]
                [keechma/entitydb "0.1.3"]
                [keechma/toolbox "0.1.17" :exclusions [keechma cljsjs/react cljsjs/react-dom]]

                ]

 ;; set an nrepl port for connection to a REPL.
 :nrepl
 {:middleware
  [sc.nrepl.middleware/wrap-letsc
   refactor-nrepl.middleware/wrap-refactor]}

 :cache-blockers #{learning.versioning}

 :builds
 {:tests
  {:target :browser-test
   :test-dir "js/compiled_tests"
   :devtools
   {:http-port 8606
    :http-root "js/compiled_tests"}}
  :ci
  {:target :karma
   :output-to "js/compiled_karma/tests.js"
   :ns-regexp "devcards$"}

  :node
  {:target :node-script
   :output-to "scripts/compiled/tools.js"
   :main learning.tools.core/main}

  :app
  {:target :browser
   :output-dir "public/js/compiled"
   :asset-path "/js/compiled"
   :closure-defines {learning.web.web-worker.core/ASSET-PATH "/js/compiled"}
   :module-hash-names true
   :devcards true
   :modules
   {:main {:entries [learning.web.core]}
    :worker {:entries [learning.web.web-worker.worker]
             :depends-on #{:main}
             :web-worker true}}
   :compiler-options {:closure-warnings {:global-this :off}
                      :externs ["externs/bluebird.js"
                                "externs/process.js"]
                      :cache-analysis false}
   :release {:compiler-options {:elide-asserts false}}
   :devtools
   {:repl-pprint true
    :before-load learning.web.core/stop
    :after-load learning.web.core/start
    :http-root    "public"
    :http-port    8700
    :preloads     [devtools.preload learning.web.dev-requires]
    }}

  :devcards
  {:target :browser
   :output-dir "public/js/compiled_devcards"
   :asset-path "/js/compiled_devcards"
   :devcards true
   :closure-defines {learning.web.web-worker.core/ASSET-PATH "/js/compiled_devcards"}
   :js-options
    {:resolve {"isomorphic-git" {:target :global
                                 :global "git"}}}

   :modules
   {:main {:entries [learning.web.devcards]}
    :worker {:entries [learning.web.web-worker.worker]
             :depends-on #{:main}
             :web-worker true}}
   :compiler-options {:closure-warnings {:global-this :off}
                      :devcards true}
   :devtools
   {:before-load learning.web.devcards/stop
    :after-load learning.web.devcards/start
    ;; serve the public directory over http at port 8700
    :http-root    "public"
    :http-port    8701
    ;;:http-handler shadow.http.push-state/handle
    :preloads     [devtools.preload]
    }}}}
```

**Comment by mjmeintjes** [comment]
And I run `shadow-cljs watch devcards`

**Comment by thheller** [comment]
Do you use the `asn1.js` package directly or is it included via some other JS dep?

**Comment by thheller** [comment]
It should be fixed in `2.4.12`.

**Comment by mjmeintjes** [comment]
I'm also not sure why cache-analysis is in there.

The externs are due to naming conflicts in release builds that happened a
while back, adding the externs seemed to fix the problem.
